### PR TITLE
Hunting Queries KQL validations for Multiple Solutions

### PR DESCRIPTION
**Pull request by v-vdixit**
Required items, please complete
   
   Change(s):
   - Fixed failing kql validations in multiple solutions Hunting Queries

   Reason for Change(s):
   - Failing kql validation tests

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Solutions/Oracle Cloud Infrastructure/Hunting Queries/OCIDestinationsOut.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ query: |
   | where data_action_s =~ 'ACCEPT'
   | where ipv4_is_private(SrcIpAddr)
   | where ipv4_is_private(DstIpAddr) == False
-  | summarize count() by SrcIpAddr, DstIpAddr, DstPortNumber
+  | summarize count() by SrcIpAddr, DstIpAddr, DstPortNumber, data_sourceAddress_s
```
Comment (devikamehra) [inline Solutions/Oracle Cloud Infrastructure/Hunting Queries/OCIDestinationsOut.yaml:21]: Remove data_sourceAddress_s and map IPCustomEntity = SrcIpAddr.

File: Solutions/OracleDatabaseAudit/Hunting Queries/OracleDBAuditDbConnectNonOperationalTime.yaml [hunk 2]
```diff
@@ -24,7 +24,7 @@ query: |
   | where isnotempty(DstUserName)
   | extend day_of_week = dayofweek(TimeGenerated)
   | extend hour_of_day = hourofday(TimeGenerated)
-  | where day_of_week in~ ('0.00:00:00', '6.00:00:00') or hour_of_day between (18 .. 8)
+  | where tostring(day_of_week) in~ ('0.00:00:00', '6.00:00:00') or hour_of_day between (18 .. 8)
```
Comment (devikamehra) [inline Solutions/OracleDatabaseAudit/Hunting Queries/OracleDBAuditDbConnectNonOperationalTime.yaml:27]: In addition to this, hour_of_day between (8 .. 18)

File: Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareDomains.yaml [hunk 3]
```diff
@@ -14,7 +14,7 @@ relevantTechniques:
 query: |
   GCPCloudDNS
   | where TimeGenerated > ago(24h)
-  | summarize count() by Query
+  | summarize count() by Query, SrcIpAddr
```
Comment (devikamehra) [inline Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareDomains.yaml:17]: In this case also, remove the SrcIpAddr as this hunting query is only looking for DNS query alone.

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceRareDocType.yaml [hunk 4]
```diff
@@ -17,7 +17,7 @@ query: |
   | where isnotempty(DocType)
   | summarize users = makeset(ActorEmail), count() by DocType
   | top 20 by count_ asc 
-  | extend AccountCustomEntity = Users
+  | extend AccountCustomEntity = users
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceRareDocType.yaml:20]: Hi @devikamehra I have added 'users' parameter from line 18 only in the query, thanks!

File: Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareErrors.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ query: |
   GCPCloudDNS
   | where TimeGenerated > ago(24h)
   | where EventResultDetails != 'NOERROR'
-  | summarize count() by EventResultDetail, Query
+  | summarize count() by EventResultDetails, Query, SrcIpAddr
```
[the remark this answers is not in the document]
Comment (v-vdixit) [reply Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareErrors.yaml:18]: Once we remove SrcIpAddr from line 18, the query is unable to identify SrcIpAddr column being used in line 21 of the query, please suggest a fix for this, thanks!

[the remark this answers is not in the document]
Comment (devikamehra) [reply Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareErrors.yaml:18]: I understand that would happen. We would have to remove it from mapping at line 21 also.

File: Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceRareDocType.yaml [hunk 6]
```diff
@@ -17,7 +17,7 @@ query: |
   | where isnotempty(DocType)
   | summarize users = makeset(ActorEmail), count() by DocType
```
Comment (devikamehra) [inline Solutions/GoogleWorkspaceReports/Hunting Queries/GWorkspaceRareDocType.yaml:18]: Use make_set instead

**Review by devikamehra — CHANGES_REQUESTED** [review]

File: Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareDomains.yaml [hunk 7]
```diff
@@ -17,7 +17,7 @@ query: |
   | summarize count() by Query
   | order by count_ asc
   | top 10 by count_
-  | extend DNSCustomEntity = Query, IPCustomEntity = SrcIpAddr
+  | extend DNSCustomEntity = Query
 entityMappings:
   - entityType: DNS
     fieldMappings:
```
Comment (devikamehra) [inline Solutions/GoogleCloudPlatformDNS/Hunting Queries/GCPDNSRareDomains.yaml:23]: Remove IP from field mapping as well. This is applicable to all the queries.

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Review by devikamehra — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**